### DISCUSSION TITLE
Does setting focal length allow divergent beams from ring source?

### DISCUSSION THREAD

**Discussion by RecurvedBow** [fangq/mcx#223] [Help]
Hello, I have the following problem: I defined a ring source with an inner radius r_i and an outer radius r_o. Based on https://github.com/fangq/mcx/pull/182#issuecomment-170916615, I want to set a focal length f from an angle alpha (in radians) so that the ring source generates divergent beams with an angle a/2 on each side, so something like this:
![266074484-07bb8bf0-94e1-4b4d-a7f2-0529784e08f4](https://github.com/fangq/mcx/assets/101172476/1518a187-fcb3-4703-a7fe-14caf7e70aa5)

Therefore, I set the focal length to be: f = -(r_o - r_i) / tan(alpha / 2) and set cfg.srcdir(4) = f. However, resulting from this, I rather seem to get something like this instead:
![actual](https://github.com/fangq/mcx/assets/101172476/8f1c386d-68a5-4602-a9c3-d04420748b8d)

Based on the results:
![image](https://github.com/fangq/mcx/assets/101172476/50bcdf50-d46f-4e0c-90cd-99eaca35274c)

Thanks for the help in advance.

**Comment by fangq** [comment]
@RecurvedBow, the focal point of an area source (which ring source is one) is located on the axis perpendicular to the area and passing by the center of the source. The signed focal length is basically the distance from the intersection point of the axis to the source aperture, positive means converting beam, negative means diverging.

however, for disk/ring/Gaussian sources, the axis passes by the center of the circle determined by cfg.srcpos, even in the case of ring it can be asymmetrical.

https://github.com/fangq/mcx/blob/v2024.2/src/mcx_core.cu#L1361-L1385

**Reply by RecurvedBow** [reply]
Thank you for your answer! It greatly helped me understand the behavior of the ring source with focal length. However, in that case, I want to ask whether it is possible to achieve the "Desired behavior" illustrated below, on which beam divergence is applied to the inner circle of the ring in the opposite direction of the actual behavior with focal length. 

![2024-06-12_10-39](https://github.com/fangq/mcx/assets/101172476/7e8f7643-e450-46f5-adf1-1260c8e7fd01)

**Comment by fangq** [comment]
@RecurvedBow, if you set the focal length parameter to **nan**, every point on the source launches photons isotropically in space (like an isotropic point source), this will give you a highly divergent source, although perhaps too divergent

if you set the focal length to **-inf**, the launch angular profile follows the Lambertian (cosine) distribution. you can adjust the distance to surface to control the spreading.

https://github.com/fangq/mcx/blob/v2024.2/mcxlab/mcxlab.m#L76-L79

but if you want to create a finite angle spreading profile like you drew, you will need to use the customized launchangle control like I showed in this example script

https://x.com/FangQ/status/1711249367812706445
